internal/channel/channel.go

```diff
@@ -27,6 +32,31 @@ type Channel struct {
 	pluginCtxCancel func()
 }
 
+func (c *Channel) MarshalLogObject(encoder zapcore.ObjectEncoder) error {
+	encoder.AddInt64("id", c.ID)
+	encoder.AddString("name", c.Name)
+	encoder.AddString("type", c.Type)
+	encoder.AddTime("changed_at", c.ChangedAt.Time())
+	encoder.AddBool("deleted", c.IsDeleted())
+	return nil
+}
+
+func (c *Channel) GetID() int64 {
+	return c.ID
+}
+
+func (c *Channel) GetChangedAt() types.UnixMilli {
+	return c.ChangedAt
+}
+
+func (c *Channel) IsDeleted() bool {
+	return c.Deleted.Valid && c.Deleted.Bool
+}
```

Comment:
I thought about adding some `struct` type to be embedded elsewhere, but this would lead to cyclic imports unless I would locate this new type somewhere new. Tomorrow, I am going to play around with this idea a bit more.

Reply:
I broke this down even further by splitting the `Validate` method off into its own interface, as almost no type required an implementation therefore.

https://github.com/Icinga/icinga-notifications/blob/b746738576303c41e43789e36fed6e5295848f45/internal/config/incremental_sync.go#L28

Reply:
Thanks for this suggestion, @yhabteab. I have given this a try in my implementation.